tests/baselines/reference/deleteExpressionMustBeOptional_exactOptionalPropertyTypes(exactoptionalpropertytypes=true).errors.txt

```diff
@@ -0,0 +1,65 @@
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(25,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(26,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(27,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(28,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(29,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(30,8): error TS2790: The operand of a 'delete' operator must be optional.
+tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts(34,10): error TS2339: Property 'j' does not exist on type 'Foo'.
+
+
+==== tests/cases/compiler/deleteExpressionMustBeOptional_exactOptionalPropertyTypes.ts (7 errors) ====
+    interface Foo {
+        a: number
+        b: number | undefined
+        c: number | null
+        d?: number
+        e: number | undefined | null
+        f?: number | undefined | null
+        g: unknown
+        h: any
+        i: never
+    }
+    
+    interface AA {
+        [s: string]: number
+    }
+    
+    type BB = {
+        [P in keyof any]: number
+    }
+    
+    declare const f: Foo
+    declare const a: AA
+    declare const b: BB
+    
+    delete f.a
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.b
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.c
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.d
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.

```

Comment:
This was a bug - we need to check whether the property is optional.

https://github.com/microsoft/TypeScript/issues/44852

Reply:
Should `f.f` be ignored too? Or am I wrong?